pkgs/tools/system/uefitool/common.nix

```diff
@@ -0,0 +1,37 @@
+{ version, sha256, installFiles }:
+{ lib, pkgs, stdenv, fetchFromGitHub, qtbase, qmake, cmake, zip }:
+
+stdenv.mkDerivation rec {
+  passthru = {
+    inherit version;
+    inherit sha256;
+    inherit installFiles;
+  };
+  name = "uefitool-${version}";
+
+  src = fetchFromGitHub {
+    inherit sha256;
+    owner = "LongSoft";
+    repo = "uefitool";
+    rev = version;
+  };
+
+  buildInputs = [ qtbase ];
+  nativeBuildInputs = [ qmake cmake zip ];
+
+  configurePhase = ":";
+  buildPhase = "bash unixbuild.sh";
+
+  installPhase = ''
+    mkdir -p "$out"/bin
+    cp ${lib.concatStringsSep " " installFiles} "$out"/bin
+  '';
+
+  meta = with stdenv.lib; {
+    description = "UEFI firmware image viewer and editor";
+    homepage = https://github.com/LongSoft/uefitool;
+    license = licenses.bsd2;
+    maintainers = with maintainers; [ ajs124 ];
+    platforms = platforms.all;
```

Comment:
Should just be linux only i think: https://github.com/NixOS/nixpkgs/pull/45963#issuecomment-435219043

Reply:
#45963 use a quite different approach to building the project. I opted for simply executing the build script upstream distributes, which does support darwin.
Although looking at the script, seems like it manually checks for qt and tries to download it, if it doesn't exist, so it probably won't work.
Ok, I'll mark it and add a comment.